examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairingCommand.kt

```diff
@@ -0,0 +1,253 @@
+    networkType = network
+    filterType = filter
+    try {
+      remoteAddr = IPAddress(InetAddress.getByName("::1"))
+    } catch (e: UnknownHostException) {
+      throw RuntimeException(e)
+    }
+    addArgument("node-id", 0, Long.MAX_VALUE, nodeId, null, false)
+    when (networkType) {
+      PairingNetworkType.NONE -> {}
+      PairingNetworkType.WIFI -> {
+        addArgument("ssid", SSID, null, false)
+        addArgument("password", password, null, false)
+      }
+
+      PairingNetworkType.THREAD -> addArgument("operationalDataset", operationalDataset, null, false)
+    }
+    when (mode) {
+      PairingModeType.NONE -> {}
+      PairingModeType.CODE, PairingModeType.CODE_PASE_ONLY -> {
+        addArgument("payload", onboardingPayload, null, false)
+        addArgument("discover-once", discoverOnce, null, false)
+        addArgument("use-only-onnetwork-discovery", useOnlyOnNetworkDiscovery, null, false)
+      }
+
+      PairingModeType.ADDRESS_PASE_ONLY -> {
+        addArgument("setup-pin-code", 0, 134217727, setupPINCode, null, false)
+        addArgument("device-remote-ip", remoteAddr, false)
+        addArgument("device-remote-port", 0.toShort(), Short.MAX_VALUE, remotePort, null, false)
+      }
+
+      PairingModeType.BLE -> {
+        addArgument("setup-pin-code", 0, 134217727, setupPINCode, null, false)
+        addArgument("discriminator", 0.toShort(), 4096.toShort(), discriminator, null, false)
+      }
+
+      PairingModeType.ON_NETWORK -> addArgument("setup-pin-code", 0, 134217727, setupPINCode, null, false)
+      PairingModeType.SOFT_AP -> {
+        addArgument("setup-pin-code", 0, 134217727, setupPINCode, null, false)
+        addArgument("discriminator", 0.toShort(), 4096.toShort(), discriminator, null, false)
+        addArgument("device-remote-ip", remoteAddr, false)
+        addArgument("device-remote-port", 0.toShort(), Short.MAX_VALUE, remotePort, null, false)
+      }
+
+      PairingModeType.ALREADY_DISCOVERED -> {
+        addArgument("setup-pin-code", 0, 134217727, setupPINCode, null, false)
+        addArgument("device-remote-ip", remoteAddr, false)
+        addArgument("device-remote-port", 0.toShort(), Short.MAX_VALUE, remotePort, null, false)
+      }
+    }
+    when (filterType) {
+      DiscoveryFilterType.NONE -> {}
+      DiscoveryFilterType.SHORT_DISCRIMINATOR -> addArgument(
+        "discriminator",
+        0.toShort(),
+        4096.toShort(),
+        discriminator,
+        null,
+        false
+      )
+
+      DiscoveryFilterType.LONG_DISCRIMINATOR -> addArgument(
+        "discriminator",
+        0.toShort(),
+        4096.toShort(),
+        discriminator,
+        null,
+        false
+      )
+
+      DiscoveryFilterType.VENDOR_ID -> addArgument(
+        "vendor-id",
+        0,
+        Long.MAX_VALUE,
+        discoveryFilterCode,
+        null,
+        false
+      )
+
+      DiscoveryFilterType.COMPRESSED_FABRIC_ID -> addArgument(
+        "fabric-id",
+        0,
```

Comment:
Shouldn't this be `Long.MIN_VALUE` to support the full 64-bit range?

Reply:
I don't think "fabric-id" can be a negative number

Reply:
Since Java `long` types are signed, the upper bit is used as the sign bit, so for larger fabric ID values you can run into negative values. For example https://pl.kotl.in/SxOlTnKPv

> FDAB010122223333 is -168039456716999885 as a Java long.